CHANGELOG.md

```diff
@@ -0,0 +1,13 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+# Unreleased
+
+N/A
+
+# Release 1.0.0
```

Comment:
Well, if the SO version is now 3.0. I would argue that this is now also Release 3.0. This is the whole point of this versioning scheme…

Reply:
@neverpanic requesed to change it to 1.0.0  when i had release 3.0.0. I agree that this could be confusing, on the other hand this would mean that we have to major release on ABI breaking changes (But then i would also put this in releasing requirements). Taking ABI into account was also requested https://github.com/semver/semver/issues/162 at semver years ago, but has not beed added (not even optionally) @garw @neverpanic What to do?

Reply:
@garw As I understand semver, it considers API only, but API and ABI can sometimes have breaking changes independent from each other.

For example, if we change the value of a constant in a header, or re-order an enum, that would break ABI, but the API would stay the same. Since the API didn't change, according to semver, the version number should not be increased -- however, the SOVERSION would have to.

For this reason, I would have decoupled SOVERSION from the library version number according to semver. Opinions?

Reply:
Well, I don't think that we have to be very religious with the semantic versioning idea. We should just feel free to also increase the version if the ABI breaks even if this is not strictly what the semantic versioning says but it is what (imho) makes sense and also matches the intention of semantic versioning.

So maybe we just put a comment in that we will not strictly follow the rules but will increase the version number also in case of ABI incompatible changes. I see no benefit in sticking to the rules just for the purpose of sticking to the rules.

As an advantage, we could have SO-version = library version - which from my perspective is something that simplifies a lot of things and reduces confusion.